src/libraries/System.Console/src/System/Console.cs

```diff
@@ -119,9 +145,37 @@ public static ConsoleKeyInfo ReadKey(bool intercept)
             return ConsolePal.ReadKey(intercept);
         }
 
-        public static TextWriter Out => EnsureInitialized(ref s_out, () => CreateOutputWriter(OpenStandardOutput()));
+        public static TextWriter Out
+        {
+            get
+            {
+                return Volatile.Read(ref s_out) ?? EnsureInitializedOut();
 
-        public static TextWriter Error => EnsureInitialized(ref s_error, () => CreateOutputWriter(OpenStandardError()));
+                static TextWriter EnsureInitializedOut()
```

Comment:
I like the pattern with the local methods used for the initialization. 

Nit: Would it make sense to call all the local methods just `EnsureInitialized` ? Some of the names are getting too long.